src/Table.cpp

```diff
@@ -1024,4 +1015,40 @@ namespace galsim {
         _pimpl->gradientGrid(xvec, yvec, dfdxvec, dfdyvec, Nx, Ny);
     }
 
+    void WrapArrayToPeriod(double* x, int n, double x0, double period)
+    {
+#ifdef __SSE2__
+        for (; n && !IsAligned(x); --n)
+            *x -= period * floor((*x-x0)/period);
+
+        int n2 = n>>1;
+        int na = n2<<1;
+        n -= na;
+
+        if (n2) {
+            __m128d xx0 = _mm_set1_pd(x0);
+            __m128d xperiod = _mm_set1_pd(period);
+            __m128d xzero = _mm_set1_pd(0.);
+            __m128d* xx = reinterpret_cast<__m128d*>(x);
+            do {
+                __m128d xoffset = _mm_sub_pd(*xx, xx0);
+                __m128d nperiod = _mm_div_pd(xoffset,xperiod);
+                __m128d floornp = _mm_cvtepi32_pd(_mm_cvttpd_epi32(nperiod));
```

Comment:
FWIW, I tried using _mm_floor_pd from SSE4.1 here, but it didn't make things any faster...  :(

Reply:
Yeah, I was kind of disappointed that this didn't have a bigger impact.  I thought even just going to C++ for this would have been a big help.  But somehow the pure numpy thing was already close to as fast as it could get.